bin/build

```
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+# fail fast
+#set -eo pipefail
+
+set -eu
+
+start_time=$(date +%s%N)
+
+# debug
+if [[ -v BUILDPACK_APT_DEBUG ]]; then
+  set -x
+fi
+
+# shellcheck disable=SC1090,SC1091
+. "$(dirname "${BASH_SOURCE[0]}")/util.sh"
+
+layers_dir="$1"
+env_dir="$2/env"
+
+if compgen -G "${env_dir}/*" > /dev/null; then
+  for var in "${env_dir}"/*; do
+    declare "$(basename "${var}")=$(<"${var}")"
+  done
+fi
+
+apt_layer=$layers_dir/apt
+mkdir -p "$apt_layer"
+
+# Store which STACK we are running on in the cache to bust the cache if it changes
+if [ -f "$apt_layer"/STACK ]; then
+  CACHED_STACK=$(cat "$apt_layer/STACK")
+else
+  CACHED_STACK=$CNB_STACK_ID
+fi
+
+# Ensure we store the STACK in the cache for next time.
+echo "$CNB_STACK_ID" > "$apt_layer/STACK"
```

```suggestion
echo -n "$CNB_STACK_ID" > "$apt_layer/STACK"
```

stop `echo` from adding a trailing `\n`